index.js

```diff
@@ -497,18 +498,23 @@ module.exports = function (log, indexesPath) {
     const indexNeedsUpdate = !coreIndexNames.includes(op.data.indexName)
 
     function save(count, offset) {
-      if (updatedSeqIndex) saveCoreIndex('seq', indexes['seq'], count)
+      const done = multicb({ pluck: 1 })
+      if (updatedSeqIndex) saveCoreIndex('seq', indexes['seq'], count, done())
 
       if (updatedTimestampIndex)
-        saveCoreIndex('timestamp', indexes['timestamp'], count)
+        saveCoreIndex('timestamp', indexes['timestamp'], count, done())
 
       if (updatedSequenceIndex)
-        saveCoreIndex('sequence', indexes['sequence'], count)
+        saveCoreIndex('sequence', indexes['sequence'], count, done())
 
       index.offset = offset
       if (op.data.version > index.version) index.version = op.data.version
 
-      if (indexNeedsUpdate) saveIndex(op.data.indexName, index, count)
+      if (indexNeedsUpdate) {
+        done(() => {
```

Comment:
Just a doubt: is it possible that none of the core indexes would need updates? And what happens with the done then?

Reply:
That works, have tested :)